app/main.py

```diff
@@ -1,2 +1,132 @@
+from typing import Any, Hashable
+from fractions import Fraction
+from app.node import Node
+
+
 class Dictionary:
-    pass
+    def __init__(self) -> None:
+        self._length: int = 0
+        self._hash_table: list = [None] * 8
+        self._hash_size: int = 8
+        self._resize_factor: Fraction = Fraction(2, 3)
+        self._resize_multiplier: int = 2
+
+    def __setitem__(self, key: Hashable, value: Any) -> None:
+        self._hash_size_check_increase()
+        key_exists, key_hash_position = self._while_key_hash_occupied(
+            action="set", key=key, value=value
+        )
+        if not key_exists:
+            node = Node(hash_=hash(key), key=key, value=value)
+            self._hash_table[key_hash_position] = node
+            self._length += 1
+
+    def __getitem__(self, key: Hashable) -> Any | None:
+        key_exists, return_value = self._while_key_hash_occupied(
+            action="get", key=key
+        )
+        if key_exists:
+            return return_value
+        raise KeyError("Key is not found in the Dictionary")
+
+    def __len__(self) -> int:
+        return self._length
+
+    def __repr__(self) -> str:
+        result = [f"{item.key}:{item.value}"
+                  for item in self._hash_table
+                  if item is not None]
+        return "{" + ", \n".join(result) + "}"
+
+    def __delitem__(self, key: Hashable) -> None:
+        """
+        func to delete item by key if key exists, otherwise raise KeyError
+        """
+        key_hash_position = self._find_key_position(key)
+        index_len = 0
+
+        while index_len <= self._hash_size:
+            if (self._hash_table[key_hash_position]
+                    and self._hash_table[key_hash_position].key == key):
+                self._hash_table[key_hash_position] = None
+                self._length -= 1
+                return
+
+            key_hash_position += 1
+            index_len += 1
+
+        raise KeyError("Key is not found in the Dictionary")
+
+    def pop(self, key: Hashable, default: Any = None) -> Any:
+        """
+        func returns/removes value/item if key exists,
+        otherwise return/raise default/KeyError
+        """
+        try:
+            result = self[key]
+            del self[key]
+        except KeyError:
+            return default
```

Comment:
Please check your logic - you should raise KeyError in case if no key and no default value